src/renderer.zig

```diff
@@ -887,8 +887,11 @@ pub const MeshSelection = struct { // MARK: MeshSelection
 	pub var selectedBlockPos: ?Vec3i = null;
 	var lastSelectedBlockPos: Vec3i = undefined;
 	var currentBlockProgress: f32 = 0;
-	var currentSwingProgress: f32 = 0;
-	var currentSwingTime: f32 = 0;
+	pub var currentSwingProgress: f32 = 0;
+	pub var hasHit: bool = false;
+	pub const swingProgressHitTime: f32 = 0.4;
```

Comment:
This has problematic side effects:
When you cancel the swing, by e.g. switching tools, you can hit the block again at 40% of the regular swing time.

Reply:
Maybe you could just give the first swing some more build up from a passive hand location, that would compensate for hitting it at the end of the animation.

We could also just lock the tool you have until the animation is over, but I'd imagine that to be rather annoying, I would like the ability to (actively) cancel an action.

Maybe @ikabod-kee has more ideas here.